rust_src/src/lisp.rs

```diff
@@ -745,6 +745,16 @@ impl LispObject {
         }
     }
 
+    // Same as the `CHECK_LIST` function.
+    #[inline]
+    pub fn list_or_error(self) -> LispObject {
```

Comment:
This function is only used in one place, and we're not using the return value :)

Generally we have `.as_foo_or_error` when we are unwrapping a value. How about defining an `.is_list` and explicitly calling `wrong_type!` in `provide`?

Reply:
I count 19, but I also thought about doing it that way. I also didn't implement `LOADHIST_ATTACH`, but that function is only used 8 times. I don't mind changing this.